### ISSUE TITLE
Failure to format documentation with trailing space

### ISSUE THREAD

**Issue by mlucool** [psf/black#1589]
**Describe the bug** A clear and concise description of what the bug is.
If there is an trailing space in documentation, black errors unless you use `--fast`

**To Reproduce** Steps to reproduce the behavior:
Note the line is `Foo ` (with a space) which causes the issue
```
def foo():
    """
    Foo 

    Returns:
        str
    """
    return "Hello"
```

Error:
```
error: cannot format test.py: INTERNAL ERROR: Black produced code that is not equivalent to the source.  Please report a bug on https://github.com/psf/black/issues.  This diff might be helpful: /tmp/blk_6bl5wft6.log
Oh no! 💥 💔 💥
1 file failed to reformat.
```

```
 $ cat /tmp/blk_6bl5wft6.log
--- src
+++ dst
@@ -15,11 +15,11 @@
       body=
         Expr(
           value=
             Constant(
               value=
-                'Foo \n\n Returns:\n str',  # str
+                'Foo\n\n Returns:\n str',  # str
             )  # /Constant
         )  # /Expr
         Return(
           value=
             Constant(
```

**Expected behavior** A clear and concise description of what you expected to happen.

**Environment (please complete the following information):**

- Version:  master (https://github.com/psf/black/commit/11f130b7deb74f424ebf57ef551953c9bc08393f)
- OS and Python version: Linux

**Does this bug also happen on master?** To answer this, you have two options:
I cannot reproduce online.

**Additional context** Add any other context about the problem here.

**Comment by JelleZijlstra** [comment]
I think #1593 fixes this.

**Comment by ichard26** [comment]
Confirmed that 5e1f620af7ff5d2bb4205e398849e5395999a0cb fixes this bug. Fix is available in version `20.08b1` (the latest as I write this):

```console
(black) ichard26@acer-ubuntu:~/programming/oss/black$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.1 LTS
Release:	20.04
Codename:	focal
(black) ichard26@acer-ubuntu:~/programming/oss/black$ python --version
Python 3.8.5
(black) ichard26@acer-ubuntu:~/programming/oss/black$ black --version
black, version 20.8b1
(black) ichard26@acer-ubuntu:~/programming/oss/black$ cat temp.py
def foo():
    """
    Foo 

    Returns:
        str
    """
    return "Hello"
(black) ichard26@acer-ubuntu:~/programming/oss/black$ black temp.py --check
would reformat temp.py
Oh no! 💥 💔 💥
1 file would be reformatted.
```

Thanks for reporting this issue @mlucool!
